sources/en.batcave/src/lib.rs

```diff
@@ -0,0 +1,288 @@
+#![no_std]
+use aidoku::{
+	alloc::{string::ToString, vec, String, Vec},
+	imports::net::Request,
+	imports::std::{parse_date, send_partial_result},
+	prelude::*,
+	Chapter, FilterValue, Home, HomeComponent, HomeLayout, ImageRequestProvider, Link, Manga,
+	MangaPageResult, MangaStatus, Page, PageContent, Result, Source,
+};
+use serde::Deserialize;
+
+const BASE_URL: &str = "https://batcave.biz";
+const REFERER: &str = "https://batcave.biz/";
+
+struct BatCave;
+
+#[derive(Deserialize)]
+struct ChapterList {
+	news_id: i32,
+	chapters: Vec<SingleChapter>,
+}
+#[derive(Deserialize)]
+struct SingleChapter {
+	date: String,
+	id: i32,
+	title: String,
+}
+#[derive(Deserialize)]
+struct PageList {
+	images: Vec<String>,
+}
+
+impl Source for BatCave {
+	fn new() -> Self {
+		Self
+	}
+
+	fn get_search_manga_list(
+		&self,
+		query: Option<String>,
+		page: i32,
+		_filters: Vec<FilterValue>,
+	) -> Result<MangaPageResult> {
+		let url = format!(
+			"{BASE_URL}/search/{}/page/{page}/",
+			query.unwrap_or_default()
+		);
+		let result = Request::get(&url)?.html()?;
+
+		let entries = result
+			.select("#dle-content > div:not(.pagination)")
+			.map(|elements| {
+				elements
+					.filter_map(|element| {
+						let cover = element.select_first("img")?.attr("abs:data-src");
+						let url = element.select_first("a")?.attr("abs:href");
+						let title = element
+							.select_first("div > h2")
+							.and_then(|x| x.text())
+							.unwrap_or_default();
+
+						Some(Manga {
+							key: url.clone().unwrap_or_default(),
+							cover,
+							title,
+							url,
+							..Default::default()
+						})
+					})
+					.collect::<Vec<Manga>>()
+			})
+			.unwrap_or_default();
+
+		let has_next_page = !entries.is_empty();
+
+		Ok(MangaPageResult {
+			entries,
+			has_next_page,
+		})
+	}
+
+	fn get_manga_update(
+		&self,
+		mut manga: Manga,
+		needs_details: bool,
+		needs_chapters: bool,
+	) -> Result<Manga> {
+		let html = Request::get(&manga.key)?.html()?;
+
+		if needs_details {
+			manga.title = html
+				.select_first("header h1")
+				.and_then(|x| x.text())
+				.unwrap_or_default();
+
+			manga.description = html.select_first(".page__text").and_then(|x| x.text());
+
+			manga.cover = html
+				.select_first(".page__poster img")
+				.and_then(|x| x.attr("abs:src"));
+
+			manga.artists = html
+				.select_first("ul > li:has(div:contains(Artist))")
+				.and_then(|x| x.text())
+				.map(|x| x.strip_prefix("Artist: ").unwrap_or_default().to_string())
+				.map(|x| vec![x.to_string()]);
```

Comment:
it looks like you're `to_string`ing twice here, so `x` should already be a `String`, right?
```suggestion
				.map(|x| x.strip_prefix("Artist: ").unwrap_or_default().to_string())
				.map(|x| vec![x]);
```

Reply:
Yes, that is correct, removed in 7eafb14

At first I thought I would be able to split this into various artists, but all examples I found used `Various` when multiple artists or writes were included.